src/PowerShellEditorServices.Engine/Services/DebugAdapter/Debugging/VariableContainerDetails.cs

```diff
@@ -0,0 +1,100 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Utility;
+
+namespace Microsoft.PowerShell.EditorServices.Engine.Services.DebugAdapter
+{
+    /// <summary>
+    /// Container for variables that is not itself a variable per se.  However given how
+    /// VSCode uses an integer variable reference id for every node under the "Variables" tool
+    /// window, it is useful to treat containers, typically scope containers, as a variable.
+    /// Note that these containers are not necessarily always a scope container. Consider a
+    /// container such as "Auto" or "My".  These aren't scope related but serve as just another
+    /// way to organize variables into a useful UI structure.
+    /// </summary>
+    [DebuggerDisplay("Name = {Name}, Id = {Id}, Count = {Children.Count}")]
+    public class VariableContainerDetails : VariableDetailsBase
+    {
+        /// <summary>
+        /// Provides a constant for the name of the Global scope.
+        /// </summary>
+        public const string AutoVariablesName = "Auto";
```

Comment:
I have seen "auto" in the variable pane of debugging. I'm not sure if that's something vscode defines that we are mirroring here or what.

Reply:
`Auto` was our attempt to mirror what the Visual Studio debugger provides where it lists the variables it thinks are most relevant in the current debug context (stack frame).  This worked better when the debugger was a traditional, non-REPL set up.  But now that we execute scripts by dot sourcing them this feature is less useful because all script level variable wind up in the global scope along with dozens of other variables.  We *could* make this work better in the dt source case by taking a "snapshot" of global variables when the session first starts, then the `Auto` would show variables not in that original "global variables" snapshot and variables that are but whose values have changed.

Reply:
When it works, it is a very useful feature.

Reply:
Yeah thats sounds like the right direction to move in. I'll call it out of scope for this PR, but if you don't mind open a quick issue to capture this I'd appreciate it!

Reply:
Done. https://github.com/PowerShell/PowerShellEditorServices/issues/1045